src/Dialect/ONNX/ONNXOps/Tensor/Resize.cpp

```diff
@@ -76,9 +76,60 @@ LogicalResult ONNXResizeOp::inferShapes(
   if (!X().getType().isa<RankedTensorType>()) {
     return success();
   }
-  Type elementType = X().getType().cast<RankedTensorType>().getElementType();
-  ONNXResizeOpShapeHelper shapeHelper(getOperation(), {});
-  return shapeHelper.computeShapeAndUpdateType(elementType);
+  // TODO : Return to this implementation once floating point scales are handled
+  //
+  // Type elementType = X().getType().cast<RankedTensorType>().getElementType();
+  // ONNXResizeOpShapeHelper shapeHelper(getOperation(), {});
+  // return shapeHelper.computeShapeAndUpdateType(elementType);
+
+  auto inputTy = X().getType().cast<RankedTensorType>();
+
+  // Output should at least has the same rank as X input
+  if (!getResult().getType().isa<RankedTensorType>()) {
+    SmallVector<int64_t, 4> dims(inputTy.getRank(), -1);
+    getResult().setType(RankedTensorType::get(dims, inputTy.getElementType()));
+  }
+
+  assert(isFromNone(scales()) != isFromNone(sizes()) &&
+         "Exactly one of scales and sizes can be defined");
+
+  // Current implementation handles constant scales only
+  if (!isFromNone(scales())) {
+    ElementsAttr scalesAttrs = getElementAttributeFromONNXValue(scales());
+    if (!scalesAttrs) {
+      return success();
+    }
+
+    SmallVector<float, 4> scalesConstant;
+    for (auto scaleAttr : scalesAttrs.getValues<FloatAttr>()) {
+      scalesConstant.emplace_back(scaleAttr.getValueAsDouble());
+    }
+
+    SmallVector<int64_t, 4> dims;
+    for (int i = 0; i < inputTy.getRank(); i++) {
+      int newDim;
+      if (ShapedType::isDynamic(inputTy.getShape()[i]))
+        newDim = -1;
+      else
+        newDim = inputTy.getShape()[i] * scalesConstant[i];
+      dims.emplace_back(newDim);
+    }
+
+    updateType(getResult(), dims, inputTy.getElementType());
+  } else {
+    ElementsAttr sizesAttrs = getElementAttributeFromONNXValue(sizes());
```

Comment:
let's keep the ONNXResizeOpShapeHelper logic in this case

Reply:
I've updated the PR with your suggestion